src/controller/progress/tracking.service.ts

```diff
@@ -36,6 +36,8 @@ import { quizBatchDto } from '../content/dto/content.dto';
 import { BootcampController } from '../bootcamp/bootcamp.controller';
 import { SubmitFormBodyDto } from './dto/form.dto';
 import { helperVariable } from 'src/constants/helper';
+import { ErrorResponse, SuccessResponse } from 'src/errorHandler/handler';
```

Comment:
Handling the Response. in the controller is better because we can call only once if it is an error Response other wise a successful response if we are going to handle it. 
example :
```
let [err, data] = await this.codingPlatformService.getSubmissionByQuestionId(questionId);
    if (err) {
      throw new  ErrorResponse(err);
    }
    return new SuccessResponse(data);
```

Reply:
@giribabu22 But how will you send appropriate message for different error or success?